modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -515,18 +515,20 @@ public CPDefinition copyCPDefinition(long cpDefinitionId)
 			cpDefinitionId);
 
 		return cpDefinitionLocalService.copyCPDefinition(
-			cpDefinitionId, cpDefinition.getGroupId());
+			cpDefinition.getUserId(), cpDefinitionId,
```

Comment:
Ehi @MahmoudAzzam you should pass the userId to the method, not use the userId from the CPDefinition, because with this implementation the author of the copy will not be who actually launched the copy but the author of the copied CPDefinition.